engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -460,6 +461,241 @@ static int GraphicsTextureTypeToImageType(int texturetype)
     return -1;
 }
 
+static dmResource::Result GetTextureDataCb(const dmResource::GetResourceDataCallbackParams* params)
+{
+    *params->m_Message = (dmGraphics::TextureImage*) params->m_UserData;
+    return dmResource::RESULT_OK;
+}
```

Comment:
I don't think I understand the reasoning behind this callback mechanic?
Is it because of the "m_Message" vs "m_Buffer" issue?